### PR TITLE
streamlined testelf, loading one workspace at a time.

### PR DESCRIPTION
**Pull request by atlas0fd00m**

### CODE REVIEW

**Review by rakuy0 — CHANGES_REQUESTED** [review]
Just a couple minor things.

Can confirm reduces memory usage for this test.

File: vivisect/tests/testelf.py [hunk 1]
```diff
@@ -22,84 +20,84 @@
...
 logger = logging.getLogger(__name__)
 
 
+def do_analyze(vw):
+    for mod in vae, vagr, vaeep, vagp:
+        try:
+            mod.analyze(vw)
+        except Exception as e:
+            import traceback
+            log.warn("ERROR in analysis module: (%r): %r", mod, e)
```
[the remark this answers is not in the document]
Comment (atlas0fd00m) [reply vivisect/tests/testelf.py]: good catch.  done.

File: vivisect/tests/testelf.py [hunk 2]
```diff
@@ -22,84 +20,84 @@
...
-    def setUpClass(cls):
-        super(ELFTests, cls).setUpClass()
 
-        cls.tests = []
-        for test in data:
+    def test_files(self):
+        results = []
+        for test in self.data:
             name, test_data, path = test
+            fname = path[-1]
```
[the remark this answers is not in the document]
Comment (atlas0fd00m) [reply vivisect/tests/testelf.py:58]: ??
i'm not sure what you mean.

[the remark this answers is not in the document]
Comment (atlas0fd00m) [reply vivisect/tests/testelf.py:58]: plus, i'm doing away with fname.  i've found a better way to identify the prefix for names starting with file.

File: vivisect/tests/testelf.py [hunk 3]
```diff
@@ -225,18 +223,12 @@ def relocs(self, vw, data, fname):
         return failed_old, failed_new
 
 
-    def names(self, vw, data, fname):
+    def names(self, vw, test_data, fname):
```
[the remark this answers is not in the document]
Comment (atlas0fd00m) [reply vivisect/tests/testelf.py]: yes!  thank you.  fixed.

File: vivisect/tests/testelf.py [hunk 4]
```diff
@@ -295,3 +287,46 @@ def test_minimal(self):
             vw = viv_cli.VivCli()
             vw.loadFromFile(fn)
 
+name_prefix_skips = [   # (prefix, MustHavePtrPrefix),
```
[the remark this answers is not in the document]
Comment (atlas0fd00m) [reply vivisect/tests/testelf.py:288]: i might agree with you, but it is only ever used here in genNames(), and doesn't serve much other purpose.  i don't see benefit of it being in helpers, but if you like i can move it there.

[the remark this answers is not in the document]
Comment (rakuy0) [reply vivisect/tests/testelf.py:288]: For now it's fine here.

**Review by rakuy0 — APPROVED** [review]

File: vivisect/tests/testelf.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rakuy0) [reply vivisect/tests/testelf.py]: Oh, I was just referring to getting rid of the fname variable and outputting the whole path, but if you've got a better way for identifying the prefix for file names, I'm cool with that too.

**Pull request merged**